### DISCUSSION TITLE
How to update the Vimg?

### DISCUSSION THREAD

**Discussion by timeanddoctor** [Kitware/trame#190] [General]
I need open a video in the trame. Are there any example to realtime update the Vimg? Or any other module to get this target, Thanks.

**Comment by jourdain** [comment]
https://www.w3schools.com/tags/tag_video.asp

**Reply by timeanddoctor** [reply]
with layout.content:
        with vuetify.VContainer(
            fluid=True,
        ):
            with html.Div():
                html.Video(width='620',
                           height='540',
                           src=CURRENT_DIRECTORY+"/01.mp4",
                           type="video/mp4")
This is my code but seems not working.
I found this :https://trame.readthedocs.io/en/latest/trame.widgets.html   
I donnot know how to use the widget of  'html'. 
Thanks.

**Reply by jourdain** [reply]
Now the missing part, is that you have to serve over http your videos and the html needs to point to that served url. 

```python
server.serve.update({
   "my_videos": CURRENT_DIRECTORY,
})

# ...

html.Video(
    width='620',
    height='540',
    src="my_videos/01.mp4",
    type="video/mp4",
)
```

**Reply by timeanddoctor** [reply]
Thanks, I have get the video success.
video=html.Video(
    width='620',
    height='540',
    src="my_videos/01.mp4",
    type="video/mp4",
)

Can I update the src?
--------------------------------------
def updateVideo():
     video.src='new.mp4'
---------------------------------
how to update this component?

**Reply by jourdain** [reply]
Yes [upload your file](https://github.com/Kitware/trame/blob/master/examples/00_howdoi/upload.py), then save it to disk and update a variable to specify the `src=` attribute to your video tag.

**Reply by timeanddoctor** [reply]
def updateVideo():
      video.src='new.mp4'

I donnot know how to update a vuetify component after the src was changed( for I didnot find there is update function similar with the 'ctrl.view_update()').

**Reply by jourdain** [reply]
Everything is explained and covered below
- [Cheatsheet](https://kitware.github.io/trame/docs/cheatsheet.html)
- [Free Course](https://kitware.github.io/trame/docs/course_intro.html)
